### ISSUE TITLE
Qiskit operation conversion ignores `Adjoint`

### ISSUE THREAD

**Issue by lillian542** [PennyLaneAI/pennylane-qiskit#396]
**Describe the bug**
When we removed the `.inv` property, we removed the line that inverted the operation, and didn't replace it with anything - so now `Adjoint` means nothing on the Qiskit devices. 

This is in the base class, so it is true across all devices, and the logic is currently also inherited by the new device prototype.

**Expected behavior**
I expect the same result (ground state) when running this circuit:

```
@qml.qnode(dev)
def circuit(x):
    qml.RX(x, 0)
    qml.adjoint(qml.RX(x, 0))
    return qml.expval(qml.PauliZ(0))    
```

regardless of whether the device is `default.qubit` or a qiskit device running on the IBMQ QASM simulator. 

**Instead, the qiskit device just applies RX twice.** I used this device vs default qubit for a minimal working example:

```
from pennylane_qiskit import IBMQCircuitRunnerDevice
from qiskit_ibm_runtime import QiskitRuntimeService, Session

service = QiskitRuntimeService(channel="ibm_quantum")
backend = service.backend("ibmq_qasm_simulator")

dev = IBMQCircuitRunnerDevice(5, provider=None, backend=backend)
```

**Comment by lillian542** [comment]
Tracked in SC as story #55173

**Comment by lillian542** [comment]
Will be fixed for the prototype in #350 
Will be fixed for master in #397
